### PR TITLE
Add basic auth

### PR DESCRIPTION
**Pull request by shubhamparikh927**
Added basic auth feature support to fhir server.
We use `BASIC_AUTH_ENABLED` environment variable to determine whether basic auth is enabled or not.
Also, username and password are passed as environment variables `BASIC_AUTH_USERNAME` and `BASIC_AUTH_PASS` respectively.

### CODE REVIEW

File: src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java [hunk 1]
```diff
@@ -143,10 +152,14 @@ private String getPatientFromToken(RequestDetails theRequestDetails) {
 	private Boolean isOAuthEnabled() {
 		return ((OAUTH_ENABLED != null) && Boolean.parseBoolean(OAUTH_ENABLED));
 	}
-
-	private Boolean isOAuthHeaderPresent(RequestDetails theRequest) {
+	
+	private Boolean isBasicAuthEnabled() {
+		return ((BASIC_AUTH_ENABLED != null) && Boolean.parseBoolean(BASIC_AUTH_ENABLED));
```
Comment (marianbuenosayres) [inline src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java]: Suggestion: In this line, in the event BASIC_AUTH_ENABLED is "true", we should check that BASIC_AUTH_USERNAME and BASIC_AUTH_PASS are set. If not, we should print a warning.

**Review by marianbuenosayres — DISMISSED** [review]
I think the changes look good. Just one comment added as a suggestion

File: src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java (hunk 1 above)
Comment (shubhamparikh927) [reply src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java]: I have updated the code can you take a look again now?

**Review by marianbuenosayres — APPROVED** [review]

File: src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java [hunk 2]
```diff
@@ -143,10 +153,20 @@ private String getPatientFromToken(RequestDetails theRequestDetails) {
 	private Boolean isOAuthEnabled() {
 		return ((OAUTH_ENABLED != null) && Boolean.parseBoolean(OAUTH_ENABLED));
 	}
-
-	private Boolean isOAuthHeaderPresent(RequestDetails theRequest) {
+	
+	private Boolean isBasicAuthEnabled() {
+		if ((BASIC_AUTH_ENABLED != null) && Boolean.parseBoolean(BASIC_AUTH_ENABLED)) {
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/CustomAuthorizationInterceptor.java:158]: Is the null check required in addition to the call to parseBoolean?

**Review by hankwallace — APPROVED** [review]
One question. Otherwise, it's good.

**Pull request merged**